smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/ShapeValueGenerator.kt

```diff
@@ -54,6 +55,13 @@ class ShapeValueGenerator(
         shape: Shape,
         params: Node,
     ) {
+        // If the node for the value to be written is NULL, then the value for this shape is not present.
+        // Just write a Swift `nil` and stop rendering.
+        if (params.type == NodeType.NULL) {
+            writer.writeInline("nil")
+            return
+        }
+
```

Comment:
We were incorrectly rendering the expected value when a collection or aggregate shape when it is supposed to be absent (i.e. "null" in the Smithy node tree for expected value.)  This if-else handles that case.